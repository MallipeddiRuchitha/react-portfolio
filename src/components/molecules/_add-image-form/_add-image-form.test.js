import React from 'react';
import { create } from "react-test-renderer";
import { render ,fireEvent} from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import AddImageForm from './_add-image-form';
//const name="image1";
const nameError="";
const urlError="";
const error=false;
describe("EditImageForm component",() =>{
    test("Matches the snapshot",() => {
        const  addImageForm= create(<AddImageForm urlError={urlError} nameError={nameError} error={error} />);
        expect(addImageForm.toJSON()).toMatchSnapshot();

    })
    test("inputField element is rendered",() => {
        const { getByTestId, getByText } = render(<AddImageForm urlError={urlError} nameError={nameError} error={error} />);
        expect(getByTestId("addImageForm")).toBeInTheDocument();
        
    });

   test("name ,url fields are rendered", () => {
        const { getAllByTestId } = render(<AddImageForm urlError={urlError} nameError={nameError} error={error} />);
       
        expect(getAllByTestId("content-input")).toHaveLength(2);

    });
   

});