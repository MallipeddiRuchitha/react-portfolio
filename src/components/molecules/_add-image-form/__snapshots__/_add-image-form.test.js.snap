// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditImageForm component Matches the snapshot 1`] = `
<div
  data-testid="addImageForm"
>
  <h3
    className="MuiTypography-root makeStyles-heading-3 MuiTypography-h3"
  >
    Add Image
  </h3>
  <form
    className="makeStyles-form-1"
  >
    <div
      className="makeStyles-input-2"
    >
      <div
        data-testid="input"
      >
        <h5
          className="MuiTypography-root MuiTypography-h5"
        >
          Name
        </h5>
        <div
          className="makeStyles-styleInputFields-5"
          data-testid="inputField"
        >
          <div
            className="MuiInputBase-root MuiInput-root MuiInput-underline"
            onClick={[Function]}
            required={true}
            style={
              Object {
                "color": "#000",
              }
            }
          >
            <input
              className="MuiInputBase-input MuiInput-input"
              data-testid="content-input"
              defaultValue=""
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={true}
              type="text"
            />
          </div>
        </div>
      </div>
      <p
        className="MuiTypography-root makeStyles-error-4 MuiTypography-body1"
      />
    </div>
    <div
      className="makeStyles-input-2"
    >
      <div
        data-testid="input"
      >
        <h5
          className="MuiTypography-root MuiTypography-h5"
        >
          Url
        </h5>
        <div
          className="makeStyles-styleInputFields-5"
          data-testid="inputField"
        >
          <div
            className="MuiInputBase-root MuiInput-root MuiInput-underline"
            onClick={[Function]}
            required={true}
            style={
              Object {
                "color": "#000",
              }
            }
          >
            <input
              className="MuiInputBase-input MuiInput-input"
              data-testid="content-input"
              defaultValue=""
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={true}
              type="text"
            />
          </div>
        </div>
      </div>
      <p
        className="MuiTypography-root makeStyles-error-4 MuiTypography-body1"
      />
    </div>
    <div
      className="makeStyles-button-6"
      data-testid="button"
    >
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
        disabled={false}
        onBlur={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          <span
            className="MuiTypography-root MuiTypography-button"
          >
             
            Add
          </span>
        </span>
      </button>
    </div>
  </form>
</div>
`;
